### PR TITLE
Dockerfile updates + misc tidying

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

- Update dockerfile to use the python `uwsgi` package instead of the alpine uwsgi server.
- Consolidate all dev requirements into the requirements-test.txt file.
- Add docs to create_static_narrative script.
- Update python requirements.
- Misc tidying up.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-X
- [ ] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (Python) I have run Ruff on changed Python code
- [ ] If appropriate, I have recompiled the app and added the updated `StaticNarrativeImpl.py`, `StaticNarrativeServer.py`, and `compile_report.json` to this PR.

# Updating Version and Release Notes (if applicable)

N/A

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,22 @@
 FROM python:3.12-alpine
 LABEL MAINTAINER KBase Developer
 
+# Install pip requirements
+COPY requirements.txt .
+COPY requirements-test.txt .
+
+# install python modules one at a time so that all deps get resolved properly
+RUN apk --update add build-base python3-dev linux-headers pcre-dev && \
```
Comment (ialarmedalien) [inline Dockerfile:9]: These extra modules are needed to compile the uwsgi python package.

File: Dockerfile [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
+    apk cache clean && \
+    pip install --upgrade pip && \
+    pip install --no-cache-dir -r requirements.txt -r requirements-test.txt && \
+    apk del build-base python3-dev linux-headers
```
Comment (ialarmedalien) [inline Dockerfile:13]: remove the extra stuff used for compilation

File: Dockerfile [hunk 3]
```diff
@@ -1,15 +1,22 @@
...
 
+# Install pip requirements
+COPY requirements.txt .
+COPY requirements-test.txt .
+
+# install python modules one at a time so that all deps get resolved properly
+RUN apk --update add build-base python3-dev linux-headers pcre-dev && \
+    apk cache clean && \
+    pip install --upgrade pip && \
+    pip install --no-cache-dir -r requirements.txt -r requirements-test.txt && \
```
Comment (ialarmedalien) [inline Dockerfile:12]: pretty excited to find out that this is possible!!

File: requirements-test.txt [hunk 4]
```diff
@@ -1,7 +1,8 @@
-coverage==7.5.4
-pytest==8.2.2
+coverage==7.6.1
+pytest==8.3.2
 pytest-cov==5.0.0
 pytest-mock==3.14.0
-pytest-recording==0.13.1
+pytest-recording==0.13.2
 requests-mock==1.12.1
+ruff==0.6.1
```
Comment (ialarmedalien) [inline requirements-test.txt:7]: only one dep in the requirements-dev file so may as well just put it in with the test requirements

**Review by briehl — APPROVED** [review]
Looks like some solid improvements!

**Pull request merged**